docs: update outdated docs for EVM RPC v2 (#3641)
dfinity/portal: docs/developer-docs/multi-chain/ethereum/evm-rpc/how-it-works.mdx

## Authentication

> Certain JSON-RPC endpoints require an API key to authenticate with the RPC provider. The EVM RPC canister is configured with the required API keys to that a developer does not need to deal with them. A developer can still choose to use custom providers (with or without API keys) to make the call instead if desired. API keys typically have a subscription fee associated with them, but developers who use the EVM RPC canister's pre-configured keys do not need to subscribe to the RPC services themselves.

Reviewer: ```suggestion
Certain JSON-RPC endpoints require an API key to authenticate with the RPC provider. The EVM RPC canister is already configured with the required API keys, although a developer can still choose to use custom providers (with or without API keys) to make the call instead if desired. API keys typically have a subscription fee associated with them, but developers who use the EVM RPC canister's pre-configured keys do not need to subscribe to the RPC services themselves.
```
